### DISCUSSION TITLE
Advice on how to build gui?

### DISCUSSION THREAD

**Discussion by Charuru** [nocodb/nocodb#5347] [Q&A] [answered]
I followed the development setup https://docs.nocodb.com/engineering/development-setup/ and made some simple UI changes, adding in my org's logo, this all works in the development version. I want to push a built version of it to my repo so I can use it as a custom npm repo.

What are the steps to correctly build my changed repo for upload? Thanks.

**Comment by Charuru** [comment]
Okay so here's what I got so far.

Making changes to nc-gui worked fine for the front-end on :3000.

Following what's in the github actions, I did npm run generate to create the static files in /dist and copied them to nc-lib-gui /dist.

This breaks all the tables giving me errors like this:

<img width="849" alt="Screen Shot 2023-03-23 at 10 21 40 AM" src="https://user-images.githubusercontent.com/214436/227233295-9893ab48-39aa-484d-bf8e-7b2d2739c291.png">

To be sure I didn't break anything I reverted all my changes and just did a generate on a clean nc-gui. What am I missing? Thanks.

**Comment by pranavxc** [comment, marked as answer]
Can you share the steps you followed?

**Reply by Charuru** [reply]
Thanks for the reply.

Starting from a clean copy of this repo.

I build SDK, Backend, Frontend.

I confirm this all to be working, both backend and frontend runs at :8080 and :3000 respectively.

Then

> cd packages/nc-gui
> npm run build:copy

Then the backend breaks with the above errors. The frontend still works fine as it doesn't use the newly generated dist.

Visually the bugs look like this.

<img width="733" alt="Screen Shot 2023-03-23 at 1 53 05 PM" src="https://user-images.githubusercontent.com/214436/227300829-decfaf7d-5c71-41ce-b595-a051a260dac8.png">

The table columns do not load.

**Reply by pranavxc** [reply]
To make it work you have to refer `nc-lib-gui` dependency locally.

```
cd packages/nocodb
npm i ../nc-lib-gui
```

**Reply by Charuru** [reply]
Thanks great, this works for the server. What's the best way to get it onto a nocodb-seed installation?
